### PR TITLE
Fix missing implict CXXRecordDecl

### LINKED ISSUE #257: [ASTImporter] Missing CXXRecordDecl in imported DeclContext (in unittests)
In ASTTests unittests there are cases where some Decls are missing during the Decl Context import.
```
 ninja ASTTests && ./tools/clang/unittests/AST/ASTTests --gtest_filter="*" 2>&1 | ag missing                                                                                                                          
warning: missing CXXRecordDecl in imported DeclContext <input.cc:1:28>
warning: missing CXXRecordDecl in imported DeclContext <input.cc:1:30>
warning: missing CXXRecordDecl in imported DeclContext <input.cc:1:30>
warning: missing CXXRecordDecl in imported DeclContext <input.cc:1:87>
warning: missing CXXRecordDecl in imported DeclContext <input.cc:1:29>
warning: missing CXXRecordDecl in imported DeclContext <input.cc:1:30>
...
```

### PR DESCRIPTION
**Pull request by martong**
Fixes #257

### CODE REVIEW

**Comment by martong** [comment]
@Xazax-hun , @gerazo Could you please take a look on this PR?

File: test/ASTMerge/class-template/test.cpp [hunk 1]
```diff
@@ -26,5 +26,5 @@ static_assert(sizeof(X0<int>().getValue(1)) == sizeof(int));
 // CHECK: class-template1.cpp:36:7: note: field 'member' has type 'int' here
 // CHECK: class-template2.cpp:36:9: note: field 'member' has type 'float' here
 
-// CHECK: 7 warnings generated.
```
Comment (Xazax-hun) [inline test/ASTMerge/class-template/test.cpp]: What is the warning that disappeared here?

Comment (martong) [reply test/ASTMerge/class-template/test.cpp]: It was `warning: missing CXXRecordDecl in imported DeclContext`

Comment (martong) [reply test/ASTMerge/class-template/test.cpp]: We addied this new warning in https://github.com/Ericsson/clang/pull/251 and https://github.com/Ericsson/clang/pull/256

**Comment by martong** [comment]
https://reviews.llvm.org/D46958

**Pull request merged**